index.js

```diff
@@ -20,35 +22,293 @@ class J5sExecutor extends Executor {
      */
     _jenkinsCommand(options, callback) {
         // To pass arguments as an array, we need to use apply
-        options.params.push(callback);
-
-        return this.jenkinsClient[options.module][options.action].apply(
+        this.jenkinsClient[options.module][options.action].apply(
             this.jenkinsClient[options.module],
-            options.params);
+            options.params.concat([callback]));
+    }
+
+    /**
+     * Create or update Jenkins job
+     * @method _jenkinsJobCreateOrUpdate
+     * @param  {String}   jobName            Jenkins job name
+     * @param  {String}   xml                Jenkins job configuration
+     * @return {Promise}
+     */
+    _jenkinsJobCreateOrUpdate(jobName, xml) {
```

Comment:
I added `@private` annotations to private methods. 

The method naming convention is from [executor-docker](https://github.com/screwdriver-cd/executor-docker/blob/master/index.js), should I remove `_`?